zilencer/views.py

```diff
@@ -106,3 +114,66 @@ def remote_server_notify_push(request,  # type: HttpRequest
                                      badge=1, zulip=apns_payload)
 
     return json_success()
+
+def register_remote_server(request):
+    # type: (HttpRequest) -> HttpResponse
+    if request.method == 'POST':
+        form = RemoteServerRegistrationForm(request.POST)
+        if form.is_valid():
+            email = form.cleaned_data['email']
+            api_key = form.cleaned_data['api_key']
+            uuid = form.cleaned_data['uuid']
+            hostname = form.cleaned_data['hostname']
+            with transaction.atomic():
+                status_obj = RemoteServerRegistrationStatus.objects.create(
+                    email=email,
+                    hostname=hostname,
+                    uuid=uuid,
+                    api_key=api_key,
+                )
+                template_prefix = 'zilencer/emails/confirm_remoteserver'
+                context = {'hostname': hostname}
+                RemoteServerConfirmation.objects.send_confirmation(
+                    status_obj,
+                    template_prefix,
+                    email,
+                    additional_context=context,
+                    host=request.get_host())
+                return HttpResponseRedirect(
+                    reverse('send_confirm', kwargs={'email': email}))
+    else:
+        form = RemoteServerRegistrationForm()
+
+    return render(request,
+                  'zilencer/register_remote_server.html',
+                  context={'form': form, 'current_url': request.get_full_path},
+                  )
```

Comment:
interesting. Even for git diffs though, a more compact style gives you more context for each diff (since you get more information in the lines above and below).
If we accept both in the codebase though, I won't argue it here.